src/dspbase.jl

```diff
@@ -696,35 +714,57 @@ function conv(u::AbstractArray{T, N},
     end
 end
 
-function conv(u::AbstractArray{<:BLAS.BlasFloat, N},
+function conv_full(u::AbstractArray{<:BLAS.BlasFloat, N},
               v::AbstractArray{<:BLAS.BlasFloat, N}) where N
     fu, fv = promote(u, v)
-    conv(fu, fv)
+    conv_full(fu, fv)
 end
 
-conv(u::AbstractArray{<:Integer, N}, v::AbstractArray{<:Integer, N}) where {N} =
-    round.(Int, conv(float(u), float(v)))
+conv_full(u::AbstractArray{<:Integer, N}, v::AbstractArray{<:Integer, N}) where {N} =
+    round.(Int, conv_full(float(u), float(v)))
 
-conv(u::AbstractArray{<:Number, N}, v::AbstractArray{<:Number, N}) where {N} =
-    conv(float(u), float(v))
+conv_full(u::AbstractArray{<:Number, N}, v::AbstractArray{<:Number, N}) where {N} =
+    conv_full(float(u), float(v))
 
-function conv(u::AbstractArray{<:Number, N},
+function conv_full(u::AbstractArray{<:Number, N},
               v::AbstractArray{<:BLAS.BlasFloat, N}) where N
-    conv(float(u), v)
+    conv_full(float(u), v)
 end
 
-function conv(u::AbstractArray{<:BLAS.BlasFloat, N},
+function conv_full(u::AbstractArray{<:BLAS.BlasFloat, N},
               v::AbstractArray{<:Number, N}) where N
-    conv(u, float(v))
+    conv_full(u, float(v))
 end
 
-function conv(A::AbstractArray{<:Number, M},
+function conv_full(A::AbstractArray{<:Number, M},
               B::AbstractArray{<:Number, N}) where {M, N}
     if (M < N)
-        conv(cat(A, dims=N)::AbstractArray{eltype(A), N}, B)
+        conv_full(cat(A, dims=N)::AbstractArray{eltype(A), N}, B)
     else
         @assert M > N
-        conv(A, cat(B, dims=M)::AbstractArray{eltype(B), M})
+        conv_full(A, cat(B, dims=M)::AbstractArray{eltype(B), M})
+    end
+end
+
+function conv_same(u::AbstractArray{T, N}, v::AbstractArray{T, N}) where {T, N}
+    su = size(u)
+    sv = size(v)
+    conv_res = conv_full(u, v)
+    if N != 1
+        conv_res[Int(floor.(sv[1]/2 + 1)):Int(floor.(sv[1]/2) + su[1]), Int(floor.(sv[2]/2 + 1)):Int(floor.(sv[2]/2) + su[2]), axes(conv_res)[3:end]...]
```

Comment:
Why only apply this to the first two dimensions?

Reply:
I always think that n-dimensional convolution only needs to be cut on the first two dimensions. 

It may take me a few days to have free time to fix it.